### PR TITLE
remove duplicated validation from pod's resourcequota admission

### PR DESCRIPTION
**Pull request by CaoShuFeng**
ResourceQuota is a validating admission plugin. Before it runs, pods
has already been validated. It's not necessary to validate it again.

**Release note**:
```release-note
NONE
```

### CODE REVIEW

File: pkg/quota/evaluator/core/pods.go [hunk 1]
```diff
@@ -118,23 +116,6 @@ func (p *podEvaluator) Constraints(required []api.ResourceName, item runtime.Obj
 		return fmt.Errorf("Unexpected input object %v", item)
 	}
 
-	// Pod level resources are often set during admission control
```
Comment (CaoShuFeng) [inline pkg/quota/evaluator/core/pods.go]: During validating admission control, nothing is changed.
So we don't need to check it the second time.

**Comment by deads2k** [comment]
Good catch.

/lgtm
/approve no-issue

**Pull request merged**